lib/kube/proxy/forwarder.go

```diff
@@ -1791,10 +1791,6 @@ func (f *Forwarder) makeSessionForwarder(sess *clusterSession) (*forward.Forward
 		if err := http2.ConfigureTransport(transport); err != nil {
 			return nil, trace.Wrap(err)
 		}
-	} else if sess.tlsConfig != nil {
-		// when certificate-authority-data is not provided in kubeconfig the tlsConfig can be nil,
-		// meaning that we will use the system default CA store.
-		sess.tlsConfig.NextProtos = nil
```

Comment:
Why did you remove this?

Reply:
Since I clone the `tls.Config` from the custom config, we do not need to reset the `NextProtos` field since it was never defined.

It was required before because the `tls.Config` was reused across multiple requests and `http2.ConfigureTransport(transport)` manipulated the `tls.Config` to setup the `h2` in `NextProtos` array which was not a valid config when using SPDY or Websockets and had to be reset.
Besides, that reset was generating a data race for concurrent connections

Reply:
tlsConfigs are generated in the following places:

- Reverse proxy/direct dial between Teleport Services

https://github.com/gravitational/teleport/blob/ff5c56dde077f7283a9dfe3ef085886dbf2e2f6e/lib/kube/proxy/forwarder.go#L1982-L1986

- Local cluster session
https://github.com/gravitational/teleport/blob/ff5c56dde077f7283a9dfe3ef085886dbf2e2f6e/lib/kube/proxy/auth.go#L177